### PR TITLE
test: add tests for announcements, dataimport, schedule apps

### PR DESCRIPTION
**Pull request by etnguyen03**
## Proposed changes
- Adds unit tests for the `announcements`, `dataimport`, and `schedule` apps.

### CODE REVIEW

File: intranet/apps/announcements/tests.py [hunk 1]
```diff
@@ -94,3 +105,371 @@ def test_announcement_request(self):
...
+    def test_approve_announcement_view(self):
+        teacher = get_user_model().objects.get_or_create(username="awilliam1", user_type="teacher", first_name="a", last_name="william")[0]
+        user = self.login()
+        request = AnnouncementRequest.objects.create(title="test", content="hello", author="Sysadmins", user=user)
+
+        self.login("awilliam1")
+
+        # The user is not in the "requested_teachers" list, so this should 302 to the index page
+        response = self.client.get(reverse("approve_announcement", kwargs={"req_id": request.id}), follow=True)
+        self.assertEqual(200, response.status_code)
```
Comment (Laur04) [inline intranet/apps/announcements/tests.py:129]: I believe you could better test that it really `302`s by first making the request without `follow=True`. So:
```
 response = self.client.get(reverse("approve_announcement", kwargs={"req_id": request.id}))
 self.assertEqual(302, response.status_code)
```
and then what you have here.

**Review by Laur04 — CHANGES_REQUESTED** [review]

File: intranet/apps/announcements/tests.py (hunk 1 above)
Comment (etnguyen03) [reply intranet/apps/announcements/tests.py:129]: The check for `response.context["messages"]` (the next line) does not work without `follow=True` and asserting a 200 response, I believe. But I'll try to see.

Comment (Laur04) [reply intranet/apps/announcements/tests.py:129]: No, that line won't work without `follow=True`. I'm suggesting leaving what you have and add the two lines I wrote above line 128.

Comment (etnguyen03) [reply intranet/apps/announcements/tests.py:129]: Oh... okay. Makes sense.

**Review by Laur04 — APPROVED** [review]
LGTM

**Pull request merged**